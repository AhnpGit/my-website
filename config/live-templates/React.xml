<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
