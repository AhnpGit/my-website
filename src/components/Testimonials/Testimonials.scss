.wrapper__testimonials {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: $white;

  .testimonials__background {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: $base-unit 0;

    background-image: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7)), url('https://lh5.googleusercontent.com/STDpIdel6vnqWYGR9BiEuDn3PznIj1bn18K6tSzQVsTQVDXKhnqRPA');
    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
    background-attachment: fixed;

    @include media-query($medium-up) {
      height: 500px;
    }
  }

  @include media-query($medium-up) {
    padding: 0 $base-unit;
  }

  .testimonials__intro {
    margin-bottom: $base-unit;

    h2 {
      text-align: center;
      word-spacing: 2px;
      letter-spacing: 1px;
      text-transform: uppercase;

      &:after {
        display: block;
        height: 2px;
        background-color: $tahitiGold;
        content: " ";
        width: 100px;
        margin: 0 auto;
        margin-top: $base-unit;
      }
    }
  }

  .testimonials__body {
    lost-center: 100% 0 flex;

    .testimonial__quote {
      lost-row: 1/3 $base-unit;

      blockquote {
        position: relative;
        font-style: italic;
        line-height: 145%;
        padding: $base-spacer;

        @include step-scale('font-size', 85%, 100%, 100%);
      }

      blockquote:before {
        content: "\201C";
        font-size: 400%;
        display: block;
        position: absolute;
        top: 0;
        left: -5px;
      }
    }

    @include media-query($medium-up) {
      padding: 0 $base-unit;
      .testimonial__quote {
        lost-column: 1/3 $base-unit;
      }
    }
  }
}

cite {
  font-size: 90%;
  margin-top: $base-spacer;
  display: block;

  img {
    height: 45px;
    border-radius: 50%;
    margin-right: $base-spacer;
    vertical-align: middle;
  }
}


