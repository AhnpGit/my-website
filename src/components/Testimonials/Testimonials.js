import React from 'react'

import Wrapper from '../../elements/Wrapper/Wrapper'
import Row from '../../elements/Row/Row'

import './Testimonials.scss'

/**
 * Testimonials component displays 3 testimonials
 */
const Testimonials = ({ testimonialItems }) => (
  <Row className="testimonials">
    <Wrapper name="testimonials">
      <div className="testimonials__background">
        <div className="testimonials__intro">
          <h2>Our customers can't live without us</h2>
        </div>
        <div className="testimonials__body">
          {testimonialItems.map(testimonial =>
            <div className="testimonial__quote" key={testimonial.id}>
              <blockquote>
                {testimonial.text}
                <cite><img src={testimonial.imageURL} alt={testimonial.customerName} />{testimonial.customerName}</cite>
              </blockquote>
            </div>
          )}
        </div>
      </div>
    </Wrapper>
  </Row>
)

export default Testimonials
