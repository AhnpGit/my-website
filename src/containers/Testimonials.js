// @flow

import { connect } from 'react-redux'
import Testimonials from '../components/Testimonials/Testimonials'
import type { TestimonialsType } from '../utils/types'

type TestimonialsState = {
  testimonials: TestimonialsType
}

export const mapStateToProps = (state : TestimonialsState) => ({
  testimonialItems: state.testimonials.testimonialItems
})

export default connect(
  mapStateToProps
)(Testimonials)
