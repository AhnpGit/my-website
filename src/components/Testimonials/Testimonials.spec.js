import React from 'react'
import { shallow } from 'enzyme'
import Testimonials from './Testimonials'

describe('Testimonials component', () => {
  it('renders correctly without any issues', () => {
    const testimonialItems = [
      {
        id: 1,
        imageURL: "https://lh6.googleusercontent.com/jD4zPA2b_nZuI-9e7Sh0HHHcRhGcJ8NK8rqWI9Z6OnsEyJC-d8R2rQ",
        customerName: "Alberto Duncan",
        text: "Omnifood is just awesome! I just launched a startup which leaves me with no time for cooking so Omnifood is a life-saver. Now that I got used to it, I couldn't live without my daily meals!"
      }
    ]

    const result = shallow(
      <Testimonials testimonialItems={testimonialItems} />
    )

    expect(result).toMatchSnapshot()
  })
})
