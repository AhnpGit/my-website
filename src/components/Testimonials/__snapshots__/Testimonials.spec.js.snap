// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials component renders correctly without any issues 1`] = `
<Row
  className="testimonials"
>
  <Wrapper
    name="testimonials"
  >
    <div
      className="testimonials__background"
    >
      <div
        className="testimonials__intro"
      >
        <h2>
          Our customers can't live without us
        </h2>
      </div>
      <div
        className="testimonials__body"
      >
        <div
          className="testimonial__quote"
        >
          <blockquote>
            Omnifood is just awesome! I just launched a startup which leaves me with no time for cooking so Omnifood is a life-saver. Now that I got used to it, I couldn't live without my daily meals!
            <cite>
              <img
                alt="Alberto Duncan"
                src="https://lh6.googleusercontent.com/jD4zPA2b_nZuI-9e7Sh0HHHcRhGcJ8NK8rqWI9Z6OnsEyJC-d8R2rQ"
              />
              Alberto Duncan
            </cite>
          </blockquote>
        </div>
      </div>
    </div>
  </Wrapper>
</Row>
`;
